core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3323,24 +3358,30 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
       }
       if (syncPlan.toLoadMetadata()) {
         loadMetadata = true;
+        pathsToLoad.add(mMountTable.getMountPoint(inodePath.getUri()));
       }
       if (syncPlan.toSyncChildren() && inode instanceof InodeDirectory
           && syncDescendantType != DescendantType.NONE) {
         UfsStatus[] listStatus = ufs.listStatus(ufsUri.toString());
-        if (listStatus != null) {
+
+        // if the alluxio path contains a mount point, we need to keep going to find it
+        if (listStatus != null || containsMountPoint) {
```

Comment:
could you explain why `containsMountPoint` is necessary here?

Reply:
Because `listStatus` works from the ufs side, and the mount points don't show up in the list. When we get a directory with just mount points, we end up skipping the whole logic of recursively calling `syncInodeMetadata` on the mount points. This is also why the previous version of the code could not pass the test added in this PR.